ring_benchmark/ring.erl

```diff
@@ -0,0 +1,56 @@
+-module(ring).
+-export([main/1]).
+
+node_setup() -> node_setup(#{}).
+node_setup(State) ->
```

Comment:
this feels a bit overly complex, it took me several reads to grok what the code does.

Particularly, I don't like that the loop handles 3 types of messages, and that the root and normal nodes have such different code